docs/data/joy/customization/introduction/ButtonThemes.js

```diff
@@ -0,0 +1,593 @@
+
+const themes = {
+  github: githubTheme,
+  fluent: fluentTheme,
+  chakra: chakraTheme,
+  mantine: mantineTheme,
+};
+const codes = {
+  github: githubCode,
+  fluent: fluentCode,
+  chakra: chakraCode,
+  mantine: mantineCode,
+};
+const prefixes = {
+  github: 'gh',
+};
+
+const useEnhancedEffect =
+  typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
+
+export default function ButtonThemes() {
+  // the `node` is used for attaching CSS variables to this demo, you might not need it in your application.
+  const [node, setNode] = React.useState(null);
+  useEnhancedEffect(() => {
+    setNode(document.getElementById('button-themes-demo'));
+  }, []);
+
+  const [design, setDesign] = React.useState('github');
+  return (
+    <CssVarsProvider
+      prefix={prefixes[design] || design}
+      theme={themes[design]}
+      colorSchemeNode={node || null}
+      colorSchemeSelector="#button-themes-demo"
+      modeStorageKey="button-themes-demo"
+      colorSchemeStorageKey="button-themes-demo"
+    >
+      <Box
+        id="button-themes-demo"
+        sx={{
+          m: -1.5,
+          mt: 0.5,
+          flexGrow: 1,
+          maxWidth: 'calc(100% + 24px)',
+          bgcolor: 'background.body',
+          '& .markdown-body pre': {
+            margin: 0,
+            borderRadius: 'xs',
+          },
+        }}
+      >
+        <Box sx={{ display: 'flex', flexDirection: 'column', gap: 1, mb: 3 }}>
+          <Box
+            sx={{
+              flexGrow: 1,
+              m: 'auto',
+              display: 'flex',
+              alignItems: 'center',
+              p: 2,
+              minHeight: 120,
+            }}
+          >
+            {design === 'github' && (
+              <Box sx={{ display: 'flex', gap: 2 }}>
+                <Button color="success">Primary</Button>
+                <Button variant="outlined" color="neutral">
+                  Default
+                </Button>
+              </Box>
+            )}
+            {design === 'fluent' && (
+              <Box sx={{ display: 'flex', gap: 2 }}>
+                <Button>Primary</Button>
+                <Button variant="outlined" color="neutral">
+                  Secondary
+                </Button>
+              </Box>
+            )}
+            {design === 'chakra' && (
+              <Box sx={{ display: 'flex', gap: 2 }}>
+                <Button>Solid</Button>
+                <Button variant="outlined">Outlined</Button>
+              </Box>
+            )}
+            {design === 'mantine' && (
+              <Box sx={{ display: 'flex', gap: 2 }}>
+                <Button>Solid</Button>
+                <Button variant="soft">Soft</Button>
+                <Button variant="outlined">Outlined</Button>
+              </Box>
+            )}
+          </Box>
+          <Box sx={{ mx: 'auto', display: 'flex', gap: 1, alignItems: 'center' }}>
+            <FormLabel htmlFor="button-theme">Theme:</FormLabel>
+            <Select
```

Comment:
Can we isolate the `Select` outside of the `CssVarsProvider`? It looks strange that it changes when the theme is changed. Also there is something wrong with the font used for the Chakra demo:

![image](https://user-images.githubusercontent.com/4512430/170457974-8d770581-b2a8-4bd1-9245-37a5918caf3c.png)